app/app.go

```diff
@@ -749,7 +758,13 @@ func New(
 		packetforwardkeeper.DefaultForwardTransferPacketTimeoutTimestamp,
 		packetforwardkeeper.DefaultRefundTransferPacketTimeoutTimestamp,
 	)
-	transferStack = delayedackmodule.NewIBCMiddleware(transferStack, app.DelayedAckKeeper)
+	middleware := delayedackmodule.NewIBCMiddleware(transferStack, app.DelayedAckKeeper, app.RollappKeeper)
+	transferStack = middleware
+	app.TransferInjectKeeper.SetMiddleware(
+		transferStack,
+		middleware,
+	)
+	transferStack = app.TransferInjectKeeper
```

Comment:
I find this section a bit hard to read
any way to restructure, rename stuff to make it more readable?

Reply:
I would stick with the `transferStack` notation. easier to read IMO

Reply:
was talking more about the middleware